Packs/GitLab/Integrations/GitLab/GitLab.yml

```diff
@@ -0,0 +1,3222 @@
+category: Utilities
+commonfields:
+  id: GitLab
+  version: -1
+configuration:
+- display: Server URL (e.g. https://gitlab.com/api/v4)
+  name: url
+  required: true
+  type: 0
+- display: API Key
+  name: api_key
+  required: true
+  type: 4
+- display: Project ID
+  name: project_id
+  required: true
+  type: 4
+- display: Trust any certificate (not secure)
+  name: insecure
+  required: false
+  type: 8
+- display: Use system proxy settings
+  name: proxy
+  required: false
+  type: 8
+description: Integration to GitLab API
+display: GitLab
+name: GitLab
+script:
+  commands:
+  - name: gitlab-version-get
+    description: Retrieve version information for this GitLab instance. Responds 200 OK for authenticated users.
+    outputs:
+    - type: String
+      contextPath: GitLab.Version.version
+      description: The version of this GitLab instance.
+    - type: String
+      contextPath: GitLab.Version.revision
+      description: The revision of this GitLab instance.
+  - name: gitlab-file-get
+    description: Allows to receive information about file in repository like name, size, content.
+    arguments:
+    - name: file_path
+      description: The file path.
+      required: false
+      isArray: false
+    - name: ref
+      description: The name of branch, tag or commit
+      required: false
+      isArray: false
+    outputs:
+    - contextPath: GitLab.File.file_name
+      description: The file name.
+      type: String
+    - contextPath: GitLab.File.file_path
+      description: The path of the file.
+      type: String
+    - contextPath: GitLab.File.size
+      description: The size of the file.
+      type: Number
+    - contextPath: GitLab.File.encoding
+      description: the encoding of the file.'
+      type: String
+    - contextPath: GitLab.File.content
+      description: File content.
+      type: String
+    - contextPath: GitLab.File.content_sha256
+      description: The file after hashing the content.
+      type: String
+    - contextPath: GitLab.File.ref
+      description: The branch the file's content was taken from.
+      type: String
+    - contextPath: GitLab.File.blob_id
+      description: The blob id.
+      type: String
+    - contextPath: GitLab.File.commit_id
+      description: The commit id of the file.
+      type: String
+    - contextPath: GitLab.File.last_commit_id
+      description: The last commit id of the file.
+      type: String
+    - contextPath: GitLab.File.execute_filemode
+      description: If the file is excute in filemode (Bool)
+      type: Boolean
+  - name: gitlab-file-create
+    description: This allows you to create a single file. File path or entry_id must be specified.
+    arguments:
+    - name: entry_id
+      description: Entry id of the file.if the user uploaded a file, he can provide the entry id. If he did, we need to read the content from the file and set the value in the file_content argument.
```

Comment:
Not sure about the second part of the description here. Why do we need it?

Reply:
entry_id or file path are required, but I wasn't sure how I can make the user that he need one of them- if I change both to "required : true  " I wouldn't be able to use only one of them here

Reply:
done